pkg/controller/machinepool/gcpactuator.go

```diff
@@ -469,38 +469,20 @@ func getNetwork(remoteMachineSets []machineapi.MachineSet,
 		return "", "", nil
 	}
 
-	remoteMachineProviderSpec, err := decodeGCPMachineProviderSpec(
+	remoteMachineProviderSpec, err := gcpprovider.ProviderSpecFromRawExtension(
 		remoteMachineSets[0].Spec.Template.Spec.ProviderSpec.Value,
-		scheme,
 	)
 	if err != nil {
 		logger.WithError(err).Warn("cannot decode GCPMachineProviderSpec from remote machinesets")
 		return "", "", errors.Wrap(err, "cannot decode GCPMachineProviderSpec from remote machinesets")
 	}
 
 	if len(remoteMachineProviderSpec.NetworkInterfaces) == 0 {
-		logger.Warn("remote machine do not have any network interfaces")
-		return "", "", errors.Wrap(err, "remote machine do not have any network interfaces")
+		logger.Error("remote machine does not have any network interfaces")
+		return "", "", errors.New("remote machine does not have any network interfaces")
 	}
 
 	network := remoteMachineProviderSpec.NetworkInterfaces[0].Network
 	subnet := remoteMachineProviderSpec.NetworkInterfaces[0].Subnetwork
 	return network, subnet, nil
 }
-
-func decodeGCPMachineProviderSpec(rawExt *runtime.RawExtension, scheme *runtime.Scheme) (*machineapi.GCPMachineProviderSpec, error) {
-	codecFactory := serializer.NewCodecFactory(scheme)
-	decoder := codecFactory.UniversalDecoder(machineapi.SchemeGroupVersion)
-	if rawExt == nil {
-		return nil, fmt.Errorf("MachineSet has no ProviderSpec")
-	}
-	obj, gvk, err := decoder.Decode([]byte(rawExt.Raw), nil, nil)
-	if err != nil {
-		return nil, fmt.Errorf("could not decode GCP ProviderSpec: %v", err)
-	}
-	spec, ok := obj.(*machineapi.GCPMachineProviderSpec)
-	if !ok {
-		return nil, fmt.Errorf("Unexpected object: %#v", gvk)
-	}
-	return spec, nil
-}
```

Comment:
I opted to just pull in the `gcpprovider` as they have `ProviderSpecFromRawExtension(rawExtension)`.